In [2]:
# import librairies
import pandas as pd
import yfinance as yf
import seaborn as sns
import matplotlib.pyplot as plt

pd.set_option('display.max_row',None)
pd.set_option('display.max_columns',None)

## API period and timerange parameter
> Use **"period"** instead of **start/end**
valid periods: **1d**,**5d**,**1mo**,**3mo**,**6mo**,**1y**,**2y**,**5y**,**10y**,**ytd**,**max**
(optional, default is '1mo')
period = "ytd" (depuis le 01 janvier année modulé) 
 

> Fetch Data by interval (including intraday if period < 60 days)
valid intervals: **1m**,**2m**,**5m**,**15m**,**30m**,**60m**,**90m**,**1h**,**1d**,**5d**,**1wk**,**1mo**,**3mo**(optional, default is '1d') interval = "1m",


- full documentation : https://pypi.org/project/yfinance/

In [10]:
cours = 'BTC-EUR'
actif = yf.Ticker(cours)
hist = actif.history(period="max",interval='1d')

## Asset definition 

In [11]:
actif.info

{'name': 'Bitcoin',
 'startDate': 1278979200,
 'description': 'Bitcoin (BTC) is a cryptocurrency launched in 2010. Users are able to generate BTC through the process of mining. Bitcoin has a current supply of 19,700,275. The last known price of Bitcoin is 67,136.17439756 USD and is up 0.41 over the last 24 hours. It is currently trading on 11047 active market(s) with $15,899,236,595.65 traded over the last 24 hours. More information can be found at https://bitcoin.org/.',
 'maxAge': 86400,
 'priceHint': 2,
 'previousClose': 61519.883,
 'open': 61519.883,
 'dayLow': 61478.383,
 'dayHigh': 61882.992,
 'regularMarketPreviousClose': 61519.883,
 'regularMarketOpen': 61519.883,
 'regularMarketDayLow': 61478.383,
 'regularMarketDayHigh': 61882.992,
 'volume': 14204839936,
 'regularMarketVolume': 14204839936,
 'averageVolume': 34573374341,
 'averageVolume10days': 24262485737,
 'averageDailyVolume10Day': 24262485737,
 'marketCap': 1215475482624,
 'fiftyTwoWeekLow': 22886.795,
 'fiftyTwoWeekHigh

## Get Top News of the asset

In [12]:
actif.news

[{'uuid': 'adf9be86-cca7-4c40-850f-885bbc5cc17f',
  'title': 'CD rates today, May 19, 2024 (up to 5.15% APY)',
  'publisher': 'Yahoo Personal Finance',
  'link': 'https://finance.yahoo.com/personal-finance/cd-rates-today-sunday-may-19-2024-100023520.html',
  'providerPublishTime': 1716112823,
  'type': 'STORY',
  'thumbnail': {'resolutions': [{'url': 'https://s.yimg.com/uu/api/res/1.2/ahgWOuX.I9DOBGc_O19Llg--~B/aD0zNDgwO3c9NTIyMDthcHBpZD15dGFjaHlvbg--/https://s.yimg.com/os/creatr-uploaded-images/2024-04/7d70aad0-f2cc-11ee-a57f-6fb602e47ca5',
     'width': 5220,
     'height': 3480,
     'tag': 'original'},
    {'url': 'https://s.yimg.com/uu/api/res/1.2/t3pG0ArnTGrwDyl8VXPRdw--~B/Zmk9ZmlsbDtoPTE0MDtweW9mZj0wO3c9MTQwO2FwcGlkPXl0YWNoeW9u/https://s.yimg.com/os/creatr-uploaded-images/2024-04/7d70aad0-f2cc-11ee-a57f-6fb602e47ca5',
     'width': 140,
     'height': 140,
     'tag': '140x140'}]}},
 {'uuid': 'a46c3c35-17e7-47bd-9669-679d72a690d3',
  'title': 'Savings interest rates today, May 1

## Get Asset Historical Value

In [18]:
hist.head()

Open        High         Low       Close  \
Date                                                                        
2014-09-17 00:00:00+00:00  359.546204  361.468506  351.586884  355.957367   
2014-09-18 00:00:00+00:00  355.588409  355.505402  319.789459  328.539368   
2014-09-19 00:00:00+00:00  328.278503  330.936707  298.921021  307.761139   
2014-09-20 00:00:00+00:00  307.665253  329.978180  303.931244  318.758972   
2014-09-21 00:00:00+00:00  318.120514  321.504517  306.502197  310.632446   

                             Volume  Dividends  Stock Splits  
Date                                                          
2014-09-17 00:00:00+00:00  16389166        0.0           0.0  
2014-09-18 00:00:00+00:00  26691849        0.0           0.0  
2014-09-19 00:00:00+00:00  29560103        0.0           0.0  
2014-09-20 00:00:00+00:00  28736826        0.0           0.0  
2014-09-21 00:00:00+00:00  20702625        0.0           0.0

### Stock Data Columns Explanation

#### 1. Date
- **Definition**: The specific date on which the stock data is recorded.
- **Example**: `2014-09-17 00:00:00+00:00`

#### 2. Open
- **Definition**: The opening price of the stock at the start of the trading day.
- **Example**: `359.55`

#### 3. High
- **Definition**: The highest price reached by the stock during the trading day.
- **Example**: `361.47`

#### 4. Low
- **Definition**: The lowest price reached by the stock during the trading day.
- **Example**: `351.59`

#### 5. Close
- **Definition**: The closing price of the stock at the end of the trading day.
- **Example**: `355.96`

#### 6. Volume
- **Definition**: The total number of shares traded during the trading day.
- **Example**: `16,389,166`

#### 7. Dividends
- **Definition**: The dividends paid to shareholders on that date, if any.
- **Example**: `0.0`

#### 8. Stock Splits
- **Definition**: The occurrence of stock splits on that date, if any.
- **Example**: `0.0`

---

#### Additional Notes:
- **Opening Price**: Indicates the first price at which a stock is traded upon the opening of an exchange on a given trading day.
- **Highest and Lowest Prices**: Help to identify the range of trading prices within the day, providing insights into the stock's volatility.
- **Closing Price**: Often used as a benchmark for the stock's daily performance and is crucial for historical comparisons.
- **Trading Volume**: Reflects the total activity and liquidity of the stock; higher volumes can indicate higher interest and market activity.
- **Dividends**: Represent a portion of the company's earnings distributed to shareholders. If zero, no dividends were paid on that date.
- **Stock Splits**: Adjust the company's number of outstanding shares and the price per share, but do not affect the company's overall market capitalization.
